pkg/platform/kube/platform.go

```diff
@@ -1199,3 +1213,53 @@ func (p *Platform) enrichTriggerWithServiceType(functionConfig *functionconfig.C
 
 	return trigger
 }
+
+func (p *Platform) validateAPIGatewayFunctionsHaveNoIngresses(platformAPIGateway *platform.APIGatewayConfig) error {
+
+	// check ingresses on every upstream function
+	errGroup, _ := errgroup.WithContext(context.TODO())
+	for _, upstream := range platformAPIGateway.Spec.Upstreams {
+		errGroup.Go(func() error {
+			function, err := p.GetFunctions(&platform.GetFunctionsOptions{
```

Comment:
another way to achieve a faster way is to 

1. prepare list of all function names
2. use label selectors
3. make the check

That way, you use one API request, receive multiple functions and iterate them
again, on a bulky system - it makes a great difference!